### PR TITLE
High AR Rebalance

### PR DESCRIPTION
**Pull request by MBmasher**
High AR is considered as AR above 10.33

Aim gets nerfed, while speed gets buffed.

Old: 
`aim_bonus = max(0.45 * (AR-10.33), 0)`

New:
```
aim_bonus = max(0.3 * (AR-10.33), 0)
speed_bonus = max(0.3 * (AR-10.33), 0)
```

### CODE REVIEW

**Comment by MBmasher** [comment]
Example Values:

[fieryrage | Lonely Go! (TV Size) [Fiery's Extreme] +HDDTHR (AR11):](https://osu.ppy.sh/b/1860433?m=0) 823.48pp before, 786.25pp after

[Matrix | Insaengui Huejeonmongma (ppongjjak ver) [Happil] +DTHR (AR11):](https://osu.ppy.sh/b/1217241) 622.34pp before, 660.56pp after

[nathan on osu | Akatsuki Zukuyo [Taeyang's Extra] +HDDT (AR10.53):](https://osu.ppy.sh/b/795627?m=0) 797.81pp before, 801.03pp after

[Mathi | Sunglow [Melody] +HDDT (AR10.53):](https://osu.ppy.sh/b/1258642) 840.83pp before, 839.91pp after

**Comment by Kverto** [comment]
I'm skeptical about the AR impacting pp, because the same execution gets awarded different values, and it is very ambigous how much the AR bonus awards reading when memorisation can play a significant role in certain cases. 

Deleting this would suffice for now in my opinion:

```
  		approachRateFactor += 0.45f * (approachRate - 10.33f);
		approachRateFactor += 0.3f * (approachRate - 10.33f);
	else if (approachRate < 8.0f)
```

**Comment by girantinas** [comment]
> 
> 
> I'm skeptical about the AR impacting pp, because the same execution gets awarded different values, and it is very ambigous how much the AR bonus awards reading when memorisation can play a significant role in certain cases.
> 
> Deleting this would suffice for now in my opinion:
> 
> ```
>   		approachRateFactor += 0.45f * (approachRate - 10.33f);
> 		approachRateFactor += 0.3f * (approachRate - 10.33f);
> 	else if (approachRate < 8.0f)
> ```

Sure but Flashlight is all about memorization and it also gives additional pp. Especially with regards to ar11 stream players, they are current undervalued, meanwhile aim-based ar 11 plays (like harumachi clover) are more prevalent and also valued more. This change aims to correct that.

**Comment by Kverto** [comment]
> Sure but Flashlight is all about memorization and it also gives additional pp.

Experienced flashlight players might argue with that, but my issue is that awarding pp for AR is very ambiguous in terms of what is actually getting rewarded, which calls for a more conservative approach in my opinion.

**Comment by Iryya** [comment]
I think this might be a little too much for AR11 streams (unko play being not an FC, with 94% acc yet it gains 155pp vs live values.)

File: src/performance/osu/OsuScore.cpp [hunk 1]
```diff
@@ -143,6 +143,13 @@ void OsuScore::computeSpeedValue(const Beatmap& beatmap)
 
 	int numTotalHits = TotalHits();
 
+	f32 approachRate = beatmap.DifficultyAttribute(_mods, Beatmap::AR);
+	f32 approachRateFactor = 1.0f;
+	if (approachRate > 10.33f)
+		approachRateFactor += 0.3f * (approachRate - 10.33f);
+
+	_speedValue *= approachRateFactor;
```
[the remark this answers is not in the document]
Comment (VINXIS) [reply src/performance/osu/OsuScore.cpp]: Yea, would also be a good idea to refactor the aim multiplier as well

**Comment by smoogipoo** [comment]
@Tom94 The osu-performance peeps see this as ready.

Here's a spreadsheet of the top 200 scores, and how they change after this: https://docs.google.com/spreadsheets/d/1fP4SyaUq7m7hIA3CGamIhzGk1xkbzqK69rMhzRk8FgA/edit?usp=sharing

Note: This data is compared against the master which has the FL changes merged in. Thus the spreadsheet shows the changes only relevant to this PR.

If you can look through the results and see if you agree with them, that'd be great.

**Comment by girantinas** [comment]
> > Here's a spreadsheet of the top 200 scores, and how they change after this: https://docs.google.com/spreadsheets/d/1fP4SyaUq7m7hIA3CGamIhzGk1xkbzqK69rMhzRk8FgA/edit?usp=sharing
> > ...
> > If you can look through the results and see if you agree with them, that'd be great.
> 
> VINXIS - Sidetracked Day **+HR** gets PP from this?
> 
> vinxpp. The other PP Rebalances nerf stream maps like Sidetracked, such as https://github.com/ppy/osu/pull/3839

**Pull request merged**